src/test/shared/telemetry/activation.test.ts

```diff
@@ -103,6 +104,7 @@ describe('isTelemetryEnabled', () => {
         it(scenario.desc, async () => {
             await settings.update('telemetry', scenario.initialSettingValue, vscode.ConfigurationTarget.Global)
 
+            await formatTelemetrySettingToBool(toolkitSettings)
```

Comment:
If you do revise the PR's impl of isTelemetryEnabled, please test formatTelemetrySettingToBool in isolation by verifying toolkitSettings directly

Reply:
Yes, please write a unit test for this.

Reply:
Great suggestion, I added a new unit test and reorganized the testing suite.